core/variant.cpp

```diff
@@ -601,6 +602,59 @@ bool Variant::can_convert_strict(Variant::Type p_type_from, Variant::Type p_type
 	return false;
 }
 
+bool Variant::can_convert(Type p_type_to) const {
+	// If this is a string variant and we are converting to a numeric type, check if the contents of the
+	// string make sense as a number
+	if (type == Type::STRING && (p_type_to == Type::INT || p_type_to == Type::REAL)) {
+		// Get string
+		String s = operator String();
+
+		// Skip first character if it is '-'
+		int i = 0;
+
+		if (s[i] == '-') {
+			i++;
+
+			// If the first character is '-', there must be at least one character afterwards
+			if (s.length() < 2) {
+				return false;
+			}
+		}
+
+		// Scan for invalid characters in whole portion of numeric literal
```

Comment:
You can use `String.is_valid_integer()` and `String.is_valid_float()` methods instead.